docs/versioned_docs/version-v1.3.0/02-Features/02-message-handling/04-customize-general.md

```diff
@@ -0,0 +1,259 @@
+---
+title: "Customize general message handling"
+layout: default
+---
+
+# Customize general message handling
+
+While the message processing is handled by the `IMessageHandler<>` implementations, the message router controls in what format the message is received.
+We allow several customizations while using a built-in or implementing your own message router.
+
+## Filter messages based on message context
+
+When registering a new message handler, one can opt-in to add a filter on the message context which filters out messages that are not needed to be processed.
+
+This can be useful when you are sending different message types on the same queue. Another use-case is being able to handle different versions of the same message type which have different contracts because you are migrating your application.
+
+Following example shows how a message handler should only process a certain message when a property's in the context is present.
+
+We'll use a simple message handler implementation:
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions.MessagingHandling;
+
+public class OrderMessageHandler : IMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, MessageContext context, ...)
+    {
+        // Do some processing...
+    }
+}
+```
+
+We would like that this handler only processed the message when the context contains `MessageType` equals `Order`.
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.WithMessageHandler<OrderMessageHandler, Order>(context => context.Properties["MessageType"].ToString() == "Order");
+    }
+}
+```
+
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
+
+## Control custom deserialization
```

Comment:
Also already mentionned in the documentation that is specific for servicebus / eventhubs.  Should we duplicate it here ?

Reply:
Or, we put that here, and refer to this part in the more specific pages ?

Reply:
It is about other extensions, but it feels the same indeed. If we move it or centralize it, it might get forgotten. We got users in the past that didn't know that it was there, despite we had it on a separate page, because it wasn't referred to on the 'specific message handling' page. So, it felt good to centralize based on 'message handling type' instead of similar extension signature.